examples/language-modeling/run_clm.py

```diff
@@ -602,10 +602,13 @@ def compute_metrics(eval_preds):
 
         metrics = train_result.metrics
 
-        max_train_samples = (
-            data_args.max_train_samples if data_args.max_train_samples is not None else len(train_dataset)
-        )
-        metrics["train_samples"] = min(max_train_samples, len(train_dataset))
+        if data_args.streaming:
+            train_samples = training_args.max_steps * training_args.per_device_train_batch_size
+        else:
+            train_samples = len(train_dataset)
+
+        max_train_samples = data_args.max_train_samples if data_args.max_train_samples is not None else train_samples
+        metrics["train_samples"] = min(max_train_samples, train_samples)
```

Comment:
Since `max_train_samples` cannot be used in streaming mode, I think we should rather have:
```suggestion
        if data_args.streaming:
            metrics["train_samples"] = training_args.max_steps * training_args.per_device_train_batch_size
        else:
            max_train_samples = data_args.max_train_samples if data_args.max_train_samples is not None else len(train_dataset)
            metrics["train_samples"] = min(max_train_samples, len(train_dataset))
```